logic/src/androidMain/kotlin/com/wire/kalium/logic/network/NetworkStateObserverImpl.kt

```diff
@@ -23,35 +23,37 @@ import android.net.ConnectivityManager
 import android.net.Network
 import android.net.NetworkCapabilities
 import com.wire.kalium.logic.kaliumLogger
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
 
 actual class NetworkStateObserverImpl(appContext: Context) : NetworkStateObserver {
     private val connectivityManager: ConnectivityManager = appContext.getSystemService(Activity.CONNECTIVITY_SERVICE) as ConnectivityManager
-    private val networkStateFlow: MutableStateFlow<NetworkState>
+    private val networkStateFlow: MutableStateFlow<Pair<NetworkState, Boolean>>
```

Comment:
comment from Vitor:

What is this boolean?
Can we extract this Pair to a class with names that would clarify that?